### PR TITLE
Test and fix listfilestrategy.py

### LINKED ISSUE #905: Error on check_md5 with folder
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

  File "\scripts\prepdocs.py", line 256, in <module>
    loop.run_until_complete(main(file_strategy, azd_credential, args))
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "\scripts\prepdocs.py", line 131, in main
    await strategy.run(search_info)
  File "\scripts\prepdocslib\filestrategy.py", line 54, in run
    async for file in files:
  File "\scripts\prepdocslib\listfilestrategy.py", line 82, in list
    if not self.check_md5(path):
  File "\scripts\prepdocslib\listfilestrategy.py", line 92, in check_md5
    with open(path, "rb") as file:
PermissionError: [Errno 13] Permission denied: 'rootPath/data\\SomeFolder'

    def check_md5(self, path: str) -> bool:
        # if path is a folder skip it
        if os.path.isdir(path):
            return True

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

Fixes #905 and adds full test coverage for relevant classes (File, LocalListFileStrategy).

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* New tests should pass
* For manual test, make dir in data and run prepdocs, it should index contents.

### CODE REVIEW

File: scripts/prepdocslib/listfilestrategy.py [hunk 1]
```diff
@@ -23,12 +23,6 @@ def __init__(self, content: IO, acls: Optional[dict[str, list]] = None):
         self.content = content
         self.acls = acls or {}
 
-    def __enter__(self):
```
Comment (pamelafox) [inline scripts/prepdocslib/listfilestrategy.py:26]: We dont seem to be using this class as a context manager anywhere, so I deleted the code.

File: tests/test_prepdocs.py [hunk 2]
```diff
@@ -1,29 +1,12 @@
...
 
-from scripts.prepdocslib.blobmanager import File
 from scripts.prepdocslib.embeddings import (
     AzureOpenAIEmbeddingService,
     OpenAIEmbeddingService,
 )
-from scripts.prepdocslib.listfilestrategy import ADLSGen2ListFileStrategy
-
-
-def test_filename_to_id():
```
Comment (pamelafox) [inline tests/test_prepdocs.py:16]: Just moved files

File: scripts/prepdocslib/listfilestrategy.py (hunk 1 above)
Comment (tonybaloney) [reply scripts/prepdocslib/listfilestrategy.py:26]: this class looks like it should mostly be used as a context manager if it has a `.close()` method, so I'm surprised it's not being used in this way

**Review by tonybaloney — APPROVED** [review]

File: scripts/prepdocslib/listfilestrategy.py (hunk 1 above)
Comment (pamelafox) [reply scripts/prepdocslib/listfilestrategy.py:26]: I think it'd be possible for it to be used that way, but it'd be a larger change affecting a few files. Right now it's used inside a try/finally, where close() is called in the finally clause.

cc @mattmsft

**Pull request merged**